badwords.txt

```diff
@@ -0,0 +1,52 @@
+buceta
+carpet muncher
+cawk
+chink
+cipa
+coon
+cyalis
+dink
+dinks
+dirsa
+dyke
+fag
+fagging
+faggitt
+faggot
+faggs
+fagot
+fagots
+fags
+fudge packer
+fudgepacker
+gaylord
+gook
+gollywog
+gaysex
+heshe
+hoar
+hoare
+hoer
+homo
+hore
+jap
+n1gga
+n1gger
+nazi
+nigg3r
+nigg4h
+nigga
+niggah
+niggas
+niggaz
+nigger
+niggers 
+retard
+shemale
+skank
+slut
+sluts
+spac
+spic
+whore
+testSwearFilter
```

Comment:
Is it wise to include a file with all these slurs etc.. when the target users for this plugin are streamers? Streamers who might get banned if they casually browse into this list while setting this up for a stream. ABP.

I would suggest instead of reinventing the wheel we should instead auto-delete if a moderator performs any ban actions on the user

Reply:
I agree, I had a look around and supposedly other people have exposed terms like this on stream but not faced action because it's been used to prevent TOS violations. It's still a bit risky though. It's not Twitch, but on Mixer I was banned for having a chat window open in an MMO where people were saying some rough stuff, so it'd also be better to not leave it to the mods as it'll still give some exposure to an off limit term. Does auto-mod cover this kind of thing on Twitch?

There are remote libraries of words we could plug into but that lacks the customisability of having your own text file which isn't ideal either. 

Another approach may be to encrypt each word and decrypt as it's loaded in as a variable. The word itself would never be seen this way and should prevent the possibility of a TOS violation entirely. But that has its own issues with adding new terms as well as the efficiency hit. Mind you it'd only be used upon launch once.